src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -94,6 +96,10 @@ public ClassFileImpl context() {
         return context;
     }
 
+    void writeMagic(int minorVersion, int majorVersion) {
```

Comment:
Can we call this `writeHeader` as this writes both the magic and the version?

Reply:
Now, through the local variable constantPool, codeSize < 325 can be achieved without extracting a separate method.